### PR TITLE
TransferBench Package

### PR DESCRIPTION
**Pull request by PedramAlizadeh**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,3 +14,35 @@ set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ..)
...
+include(ROCMCreatePackage)
+set(ROCMCHECKS_WARN_TOOLCHAIN_VAR OFF)
+
+set(PACKAGE_NAME TB)
+set(LIBRARY_NAME TransferBench)
+
+rocm_package_setup_component(${PACKAGE_NAME}
+    LIBRARY_NAME ${LIBRARY_NAME}
+    PACKAGE_NAME ${PACKAGE_NAME}
+)
```
Comment (lawruble13) [inline CMakeLists.txt]: I don't think that this is necessary, the default packages should be sufficient.

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,3 +14,35 @@ set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ..)
...
+# FILES_MATCHING
+# PATTERN "*.hpp"
+# )
+
+rocm_install(DIRECTORY src/include
+DESTINATION include/${PACKAGE_NAME}
+COMPONENT ${LIBRARY_NAME}
+FILES_MATCHING
+PATTERN "*.hpp"
+)
```
Comment (lawruble13) [inline CMakeLists.txt]: As Transferbench only has the single executable, installing the header files shouldn't be necessary. However, you do need to ensure that the executable itself is installed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -13,3 +14,35 @@ set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ..)
...
+
+rocm_create_package(
+    NAME ${LIBRARY_NAME}
+    DESCRIPTION "TransferBench package"
+    MAINTAINER "RCCL Team"
```
Comment (lawruble13) [inline CMakeLists.txt]: An email should be included in angle brackets after the maintainer name.

**Review by lawruble13 — COMMENTED** [review]
Looking pretty good, thanks Pedram! As TransferBench is fairly simple you've actually overdone it a bit here, see my comments for further information.

File: CMakeLists.txt (hunk 3 above)
Comment (PedramAlizadeh) [reply CMakeLists.txt]: Hi Lauren,
Thank you very much for the feedbacks!
I applied the modifications. Does it look correct to you?
Also, for creating the packages after "cmake .. && make", I do "make package"!
Is it the correct way to do so?
Then, I get the following error for RPM:
CPackRPM:Debug: You may consult rpmbuild logs in: 
CPackRPM:Debug:    -TransferBench-dev/build/_CPack_Packages/Linux/RPM/rpmbuildtransferbench-devel.err
CPackRPM:Debug: *** error: line 9: Illegal sequence ".." in: Version:        ..

CMake Error at /snap/cmake/1270/share/cmake-3.26/Modules/Internal/CPack/CPackRPM.cmake:1897 (message):
  RPM package was not generated!
  TransferBench-dev/build/_CPack_Packages/Linux/RPM
Call Stack (most recent call first):
  /snap/cmake/1270/share/cmake-3.26/Modules/Internal/CPack/CPackRPM.cmake:1968 (cpack_rpm_generate_package)

CPack Error: Error while execution CPackRPM.cmake

Comment (PedramAlizadeh) [reply CMakeLists.txt]: I think the problem was VERSION in the project command!
It is solved now and seems to be working!

**Review by lawruble13 — APPROVED** [review]
LGTM!

**Pull request merged**